recipes/devel/git.yaml

```diff
@@ -21,18 +21,22 @@ checkoutSCM:
     digestSHA1: "851537fc03f5a99419ef20e9b836de965c7928bd"
     extract: False
 
-buildTools: [target-toolchain, curl]
-buildVars: [PKG_VERSION, AUTOCONF_HOST,
-    CPPFLAGS, CFLAGS, LDFLAGS]
+buildTools: [target-toolchain, curl, host-toolchain]
+buildVars: [PKG_VERSION, AUTOCONF_HOST, AUTOCONF_BUILD,
+    CC, CPPFLAGS, CFLAGS, LDFLAGS]
 buildScript: |
     export CFLAGS="-I${BOB_DEP_PATHS[libs::zlib-dev]}/usr/include -I${BOB_DEP_PATHS[libs::openssl-dev]}/usr/include"
     export LDFLAGS="-L${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib -L${BOB_DEP_PATHS[libs::openssl-dev]}/usr/lib -Wl,-rpath-link=${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib -Wl,-rpath-link=${BOB_DEP_PATHS[libs::openssl-dev]}/usr/lib"
 
+    export ac_cv_fread_reads_directories=yes
+    export ac_cv_snprintf_returns_bogus=no
```

Comment:
As I've already mentioned in #142 this depends on the target libc. I don't think it's the proper solution to just hard-code this here. Once someone starts using something other than `glibc` with the basement this could easily lead to hard-to-debug problems.

Additionally, your changes hard-code these values even for native builds for no good reasons.

Reply:
hm the second point we can handle easily, so we just do the `ac_` stuff for cross-compilation. the dependencies to glibc, mh, no idea: maybe we start with that and if we are on the point for that problem, we have to adjust this?!

Reply:
> [...] maybe we start with that and if we are on the point for that problem, we have to adjust this?!

The problem I see with this approach is that users of the basement might use the `devel::git` recipe in a recipe stack with their own libc recipe. How are they supposed to know that they have to change the `devel::git` recipe in order for their setup to work properly? I mean, the compilation would still succeed and all...

Reply:
In other words: I think we'd have to ensure that compilation of `devel::git` would fail if you try to compile it with a target libc other than `glibc`...

Reply:
I would be willing to see if this really ever happens. The current state of the PR only changes the behaviour for cross compiling. This should at least not make things worse. There are other places where we set such variables to hard coded values to get the package cross compiling. So it's not unheard of and so far it worked well enough.